### PR TITLE
Added Binary Insertion Sort Algorithm 

### PR DESCRIPTION
**Pull request by cyrixninja**
# Binary Insertion Sort Algorithm 

## Description
I found that this algorithm wasn't available on the main website in Rust so I implemented it and I want to merge it with the main repo.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

File: src/sorting/binary_insertion_sort.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    for i in 1..len {
+        let key = arr[i].clone();
+        let index = binary_search(&arr[..i], &key);
+        
+        arr[index..i+1].rotate_right(1);
+        arr[index] = key;
+    }
+}
+
+fn main() {
```
Comment (siriak) [inline src/sorting/binary_insertion_sort.rs:30]: Please remove `main`, there are tests to test the implementation

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/sorting/binary_insertion_sort.rs (hunk 1 above)
Comment (cyrixninja) [reply src/sorting/binary_insertion_sort.rs]: Fixed it. I apologize for forgetting to remove the main function. I will be sure to remove main function in my future contributions

Comment (siriak) [reply src/sorting/binary_insertion_sort.rs]: No problem

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**